### PR TITLE
Projects

### PR DESCRIPTION
**Pull request by marcosgdf**

### CODE REVIEW

**Comment by marcosgdf** [comment]
Travis reports some errors by PHPCS regarding camelCase of load_board which I can't fix

File: htdocs/projet/tasks/contact.php [hunk 1]
```diff
@@ -270,204 +276,13 @@
...
-		 */
-		print '<br>';
-/*
 		// Contacts lines (modules that overwrite templates must declare this into descriptor)
 		$dirtpls=array_merge($conf->modules_parts['tpl'],array('/core/tpl'));
 		foreach($dirtpls as $reldir)
 		{
-		    $res=@include dol_buildpath($reldir.'/contacts.tpl.php');
-		    if ($res) break;
+			$res=@include dol_buildpath($reldir.'/contacts.tpl.php');
```
Comment (rdoursenaud) [inline htdocs/projet/tasks/contact.php:283]: Looks like you have mixed tabs and spaces ;)

**Comment by rdoursenaud** [comment]
Couldn't fault you apart on 2 mixed tabs and spaces lines.
Great job !

File: htdocs/projet/class/project.class.php [hunk 2]
```diff
@@ -1322,5 +1323,74 @@ function update_element($TableName, $ElementSelectId)
...
+        return false;
+    }
+
+    /**
+     *      Load indicators for dashboard (this->nbtodo and this->nbtodolate)
+     *
+     *      @param	User	$user        		Objet user
+     *      @return int         				<0 if KO, 0=Nothing to show, >0 if OK
+     */
+    public function load_board(User $user)
```
Comment (rdoursenaud) [inline htdocs/projet/class/project.class.php]: To make Travis (and me) quite happy, you could implement this into loadBoard() and keep this method as a deprecated wrapper.

File: htdocs/projet/tasks/contact.php (hunk 1 above)
Comment (marcosgdf) [inline htdocs/projet/tasks/contact.php:283]: We are supposed to use tabs instead of spaces... PHPStorm's fault probably. Should I change back to spaces?

Comment (rdoursenaud) [inline htdocs/projet/tasks/contact.php:283]: Mmm, all tabs is OK for me. But the syntactic coloration let me know that there was mixed indentation in the same line.
But don't sweat it, absolutely tiny issue.
I don't mind fixing it myself at all.
The loadBoard() would be really nice though.

**Comment by rdoursenaud** [comment]
I merged your changes to our own branch at https://github.com/GPCsolutions/dolibarr/tree/3.6-projectsenhancements.

Would your mind fixing the few issues (Missing property declarations for the most part) revealed by scrutinizer at https://scrutinizer-ci.com/g/GPCsolutions/dolibarr/inspections/c414adc2-df85-4c39-94eb-747808462633 ?

**Comment by marcosgdf** [comment]
@rdoursenaud I saw them but I'm afraid we're not allowed to do anything but fixes in any branch other than develop... property declarations are because they are not part of the class. Might do some PR to change this in the future 3.8...

**Comment by rdoursenaud** [comment]
@marcosgdf Ok, no problem. I just wanted to make sure you were aware.

**Comment by rdoursenaud** [comment]
I have two strange bugs in the "current" task page.
- It seems it displays only the first task from a project.
- It seem to only take project affectation into account, not task affectation. And I don't care about project affectations in this view…

I don't have time to look into it. Could you please have a quick glance to tell me if I'm right?

Thanks

**Comment by marcosgdf** [comment]
No, it takes task affectation but the project visibility must be set to "everybody" in order to show tasks for people not affected to the project.

I send a new PR correcting a little bug I found.

**Comment by rdoursenaud** [comment]
Thanks.

If that's how it works, the logic in this project module is completely flawed.
A fucking task contact **IS** part of the project 'cause the task **IS** what makes a project.

Sorry for the strong language, it's not directed towards you, you did an amazing job.
I just had to get it out…

I'll try to enjoy my week-end and do more testing and a bug report on monday.

**Comment by marcosgdf** [comment]
I know I know, don't worry.

I think all the module has to be reviewed because it is actually a mess just to list tasks and projects...

**Comment by rdoursenaud** [comment]
Thanks,
another weirdness we just discovered is occuring in the project and tasks alerts.

If the delay is set to 0 days, no alert is ever shown.
We expect these to show immediately with this setting.

Could you please look into it.

**Comment by marcosgdf** [comment]
Done

**Pull request closed without merging**